README.md

```diff
@@ -1,189 +1,785 @@
+    ```
+
+    This new `onCreate` does several things:
+
+    * Define the initial state for the `signInButton`.
+    * Supply authentication information for connecting to your Blockstack app: `appDomain`, `redirectURI`, `manifestURI` and `scopes`
+    * Add a listener for the button click.
+
+    Notice that the application in this example is a URI you have not set up.
+    Registering and application name takes time, so in time's interest you'll
+    use an existing app that is identical to the `hello-world` you created
+    earlier. For a produciton verison, you'll need to replace `appDomain`,
+    `redirectURI`, `manifestURI` and `scopes` with values appropriate for your
+    app.
+
+5. Add a private function to reflect when a user successfully signs in.
+
+    ```kotlin
+    private fun onSignIn(userData: UserData) {
+    		userDataTextView.text = "Signed in as ${userData.did}"
+
+    		signInButton.isEnabled = false
+
+    }
+    ```
+
+6. Handle sign in requests with an `onNewIntent` function.
+
+    Retrieve the authentication token from the custom protocol handler call and
+    send it to the Blockstack session.
+
+    ```kotlin
+    override fun onNewIntent(intent: Intent?) {
+      super.onNewIntent(intent)
+
+      if (intent?.action == Intent.ACTION_MAIN) {
+          blockstackSession().loadUserData {
+              userData -> runOnUiThread {if (userData != null) {onSignIn(userData)}}
+          }
+      } else if (intent?.action == Intent.ACTION_VIEW) {
+          handleAuthResponse(intent)
+      }
+    }
+    ```
+
+7. Create a handler for the authentication response.
+
+    ```kotlin
+    private fun handleAuthResponse(intent: Intent) {
+       val response = intent.dataString
+       if (response != null) {
+           val authResponseTokens = response.split(':')
+
+           if (authResponseTokens.size > 1) {
+               val authResponse = authResponseTokens[1]
+
+               blockstackSession().handlePendingSignIn(authResponse, { userData ->
+                   // The user is now signed in!
+                   runOnUiThread {
+                       onSignIn(userData)
+                   }
+               })
+           }
+       }
+    }
+    ```
+
+8. Instantiate the session.
+
+    ```kotlin
+    fun blockstackSession() : BlockstackSession {
+      val session = _blockstackSession
+       if(session != null) {
+          return session
+       } else {
+        throw IllegalStateException("No session.")
+      }
+    }
+    ```
+
+9. Verify your final `MainActivity.kt` code looks like this:
+
+    ```kotlin
+    class MainActivity : AppCompatActivity() {
+
+        private var _blockstackSession: BlockstackSession? = null
+
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            super.onCreate(savedInstanceState)
+            setContentView(R.layout.activity_main)
+
+            signInButton.isEnabled = false
+
+            val appDomain = URI("https://flamboyant-darwin-d11c17.netlify.com")
```

Comment:
add something like
// replace with your app domain

Reply:
At this point, I didn't want to do that with a comment, for the tutorial we don't want to do that yet which is why the discourse below the code.